src/hooks/api/zzal/useGetZzalDetails.ts

```diff
@@ -0,0 +1,12 @@
+import { useQuery } from "@tanstack/react-query";
+import { getZzalDetails } from "@/apis/zzal";
+
+const useGetZzalDetails = (imageId: number) => {
+  const { data: zzalDetails, ...rest } = useQuery({
```

Comment:
`useQuery`대신 `useSuspenseQuery`는 어떨까요?

Reply:
처음에 useSuspenseQuery를 사용했는데 모달 내부에서 적용하면 오류가 발생했습니다. 🥲
모달에서 useSuspenseQuery를 사용하면 우선 아래 에러가 뜹니다.
```
Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
```
error처리를 위해 따로 ErrorBoundary 컴포넌트를 만들어 모달을 감싸줘야하고 
감싸준 후에는 다음과 같은 에러가 뜹니다.
```
Error: A component suspended while responding to synchronous input. This will cause the UI to be replaced with a loading indicator. To fix, updates that suspend should be wrapped with startTransition.
```
startTransition로 감싸라는 에러가 뜹니다. 코드 복잡도가 불필요하게 증가하는거 같아 useQuery를 사용했습니다.

Reply:
아니면 짤 카드 클릭시 모달 외부 페이지에서 데이터를 가줘오고 모달로 전달해주는 방식은 어떨까요?

Reply:
처음에 useSuspenseQuery를 사용했는데 모달 내부에서 적용하면 오류가 발생했습니다. 🥲
모달에서 useSuspenseQuery를 사용하면 우선 아래 에러가 뜹니다.
```
Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
```
error처리를 위해 따로 ErrorBoundary 컴포넌트를 만들어 모달을 감싸줘야하고 
감싸준 후에는 다음과 같은 에러가 뜹니다.
```
Error: A component suspended while responding to synchronous input. This will cause the UI to be replaced with a loading indicator. To fix, updates that suspend should be wrapped with startTransition.
```
startTransition로 감싸라는 에러가 뜹니다. 코드 복잡도가 불필요하게 증가하는거 같아 결국에 useQuery를 사용했습니다.
다른 방법이 있을까요?